packages/synapse-react-client/src/lib/containers/markdown/MarkdownEditor.tsx

```diff
@@ -159,13 +159,15 @@ export const MarkdownEditor: React.FunctionComponent<MarkdownEditorProps> = ({
       <div>
         {currentTab === MarkdownEditorTabs.WRITE ? (
           <textarea
-            onChange={e => setText(e.target.value)}
+            onChange={e => {
+              setText(e.target.value)
+              handleTagModal(e.target.value)
+            }}
```

Comment:
When `onKeyDown` was used, the keydown event triggered the MUI dialog before the `onChange` handler updated the text (this didn't seem to be an issue with bootstrap modal). Due to how `handleUserTag` processes the text, this resulted in the `@` symbol being removed from the text after a user name or group tag was selected in the dialog.

I considered changing the event handler to `onKeyUp` instead, but typing the `@` symbol quickly resulted in a separate keyup event for each key (Shift, 2) and didn't trigger the dialog.

Instead, I moved both events into `onChange`, so we ensure that text is updated before the dialog is triggered and that dialog is triggered regardless of how quickly the user types the @ symbol.